src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -2007,11 +2007,14 @@ private BoundExpression BindSuppressNullableWarningExpression(PostfixUnaryExpres
             var expr = BindExpression(node.Operand, diagnostics);
             var type = expr.Type;
             // Report an error if there are no reference types.
-            if ((object)type != null &&
-                (object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType, (object)null, canDigThroughNullable: true) == null)
+            if ((object)type != null)
             {
-                // PROTOTYPE(NullableReferenceTypes): Should be a warning, not an error.
-                Error(diagnostics, ErrorCode.ERR_NotNullableOperatorNotReferenceType, node);
+                if ((object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType, (object)null, canDigThroughNullable: true) == null)
```

Comment:
>(object)type.VisitType((t, a, c) => t.IsErrorType() || t.IsReferenceType, (object)null, canDigThroughNullable: true) == null [](start = 20, length = 124)

Could you please elaborate what are we trying to detect with this condition?

Reply:
Please add a comment.

---
In reply to: [138423308](https://github.com/dotnet/roslyn/pull/22042#discussion_r138423308) [](ancestors = 138423308)

Reply:
Updated comment.